quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/AppManagerTest.java

```diff
@@ -0,0 +1,41 @@
+package edu.asu.diging.quadriga.core.service.impl;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.quadriga.core.data.AppRepository;
+import edu.asu.diging.quadriga.core.model.impl.App;
+import edu.asu.diging.quadriga.web.forms.AppForm;
+
+public class AppManagerTest {
+
+    @Mock
+    private AppRepository appRepo;
+
+    @InjectMocks
+    private AppManager managerToTest;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void test_addApp_success() {
+        AppForm appForm = new AppForm();
+        String name = "name";
+        String description = "testApp";
+        appForm.setName(name);
+        appForm.setDescription(description);
+        App app = managerToTest.addApp(appForm);
+        Mockito.verify(appRepo).save(app);
```

Comment:
`save` method of appRepo needs to be mocked to return an app instance with an id. you can either just make sure the app object passed into `save` is of the right type or use and argumentMatcher or argumentCaptor if you want to go beyond that.